project/lebr0nli/README.rst

```diff
@@ -0,0 +1,220 @@
+
+   Insert an item:
+
+   .. code:: python
+
+      item = {"minX": 20, "minY": 40, "maxX": 30, "maxY": 50, "data": "example"}
+      tree.insert(item)
+
+   .. epigraph::
+
+      This term project currently supports only two dimensions but aims to include three-dimensional support in the future.
+
+#. **Removing Data**
+
+   Remove a previously inserted item:
+
+   .. code:: python
+
+      tree.remove(item)
+
+   If a custom comparison is needed (e.g., comparing by specific fields), a
+   function can be provided:
+
+   .. code:: python
+
+      tree.remove(item_copy, lambda a, b: a["id"] == b["id"])
+
+   Remove all items:
+
+   .. code:: python
+
+      tree.clear()
+
+#. **Data Format**
+
+   By default, items are expected to have `minX`, `minY`, `maxX`, and `maxY`
+   properties. This format can be customized by overriding methods:
+
+   .. code:: python
+
+      class MyRBush:
+          def to_bbox(self, item):
+              return {"minX": item[0], "minY": item[1], "maxX": item[0], "maxY": item[1]}
+
+          def compare_min_x(self, a, b):
+              return a[0] - b[0]
+
+          def compare_min_y(self, a, b):
+              return a[1] - b[1]
+
+#. **Bulk-Inserting Data**
+
+   Bulk-insert data into the tree:
+
+   .. code:: python
+
+      tree.load([item1, item2, ...])
+
+   Bulk insertion improves performance compared to inserting items individually,
+   especially for clustered data.
+
+#. **Search**
+
+   Search for items within a given bounding box:
+
+   .. code:: python
+
+      result = tree.search({"minX": 40, "minY": 20, "maxX": 80, "maxY": 70})
+
+   Returns items that intersect the specified bounding box.
+
+#. **Collisions**
+
+   Check if any items intersect the given bounding box:
+
+   .. code:: python
+
+      result = tree.collides({"minX": 40, "minY": 20, "maxX": 80, "maxY": 70})
+
+   Returns `True` if there are any intersecting items, otherwise `False`.
+
+#. **Export and Import**
+
+   Export the tree data as JSON:
+
+   .. code:: python
+
+      tree_data = tree.to_json()
+
+   Import data from a previously exported JSON:
+
+   .. code:: python
+
+      tree = RBush.from_json(tree_data)
+
+   Ensure that the `max_entries` parameter is consistent between export and
+   import operations.
+
+****************************
+ Engineering Infrastructure
+****************************
+
+-  **Version Control:** The project will be open-source and hosted on GitHub.
+-  **Build System:** The build will be fully automated using `setuptools`.
+-  **Testing:** Unit tests will be provided using `pytest` to ensure correctness
+   and performance benchmarks.
+-  **Documentation:** Full documentation will be available with usage examples
+   and detailed API references.
+
+**********
+ Schedule
+**********
+
+#. **Week 1-2:** Research and analysis of `rbush` and `py-rbush`.
+#. **Week 3-4:** Prototype the core R-Tree structure and implement insertion.
+   Begin writing tests, and add deletion and search functionality.
```

Comment:
You can start writing test when researching the exiting packages, but doing it after a week or two is fine.